Simplify & Improve Build

zach2good opened the pull request:
<!-- remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm: -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

zach2good committed:
[cmake] Remove sub-library structure

zach2good committed:
[core] Improve Async tracking for Tracy

zach2good committed:
[core] Fix all the warnings thrown up by the corrected build

zach2good committed:
[core] Remove a bunch of warning exceptions and clean up their revealed warnings

zach2good committed:
[build] Start working on Clang build warnings

zach2good committed:
[build] Progress on Clang build warnings

zach2good committed:
[build] Add pch.h

claywar commented on ext/CMakeLists.txt:
```diff
@@ -24,3 +24,14 @@ CPMAddPackage(
+    efsw
+    concurrentqueue
+    PARENT_SCOPE
+)
```
Not sure if something would try to strip, but can a newline be inserted here?

claywar commented on src/map/ai/controllers/automaton_controller.cpp:
```diff
@@ -1280,7 +1275,9 @@ bool CAutomatonController::TryEnhance()
 
-    PAutomaton->StatusEffectContainer->ForEachEffect([&protect, &shell, &haste, &stoneskin, &phalanx](CStatusEffect* PStatus) {
+    // clang-format off
+    PAutomaton->StatusEffectContainer->ForEachEffect([&protect, &shell, &haste](CStatusEffect* PStatus)
```
Was there a reason for stoneskin and phalanx being removed from this?

zach2good replied:
They were warned as being unused in the scope of this lambda

claywar commented on src/map/ai/controllers/trust_controller.cpp:
```diff
@@ -488,7 +488,7 @@ uint8 CTrustController::GetPartyPosition()
 
     auto& trustList = static_cast<CCharEntity*>(POwner->PMaster)->PTrusts;
-    for (std::size_t i = 0; i < trustList.size(); ++i)
+    for (uint8 i = 0; i < trustList.size(); ++i)
```
LGTM might complain about this, since trustList.size() might return size_t

zach2good replied:
Huh, I was on autopilot when I changed that, I can't see why it would have thrown a warning

claywar commented on src/map/lua/lua_baseentity.cpp:
```diff
@@ -2890,9 +2890,7 @@ bool CLuaBaseEntity::hasTeleport(uint8 tType, uint8 bit, sol::object const& arg2
-
-    XI_DEBUG_BREAK_IF(true); // We should never get here
-    return false;
+    // return false;
```
👀

claywar commented on src/map/pch.h:
```diff
@@ -0,0 +1,75 @@
+#include "common/tracy.h"
+#include "common/xirand.h"
+
+#endif // #define _PCH_H
```
newline

claywar commented on src/map/packets/quest_mission_log.cpp:
```diff
@@ -85,7 +85,7 @@ CQuestMissionLogPacket::CQuestMissionLogPacket(CCharEntity* PChar, uint8 logID,
                 case LOG_CAMPAIGN_TWO:
                     // Second Campaign packet, summoned through logType
-                    generateCampaignMissionPacket(PChar, (uint8)256);
+                    generateCampaignMissionPacket(PChar, 1);
```
Assuming we're just enumerating the two packets now? (0,1)

zach2good replied:
That was my thought too

claywar commented on src/map/utils/charutils.cpp:
```diff
@@ -4184,7 +4187,7 @@ namespace charutils
         capacityPoints = (uint32)(capacityPoints * map_config.exp_rate);
 
-        uint16 currentCapacity = PChar->PJobPoints->GetCapacityPoints();
+        // uint16 currentCapacity = PChar->PJobPoints->GetCapacityPoints();
```
Was this unused?

zach2good replied:
Yup

WinterSolstice8 commented:
On aarch64 ubuntu 22.04, same error as `Linux_GCC10_64bit` CI

```
cat /etc/issue
Ubuntu 22.04 LTS \n \l

uname -a
Linux vps 5.15.0-1003-oracle #5-Ubuntu SMP Wed Apr 20 04:51:36 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux
```

```
c++: error: unrecognized command-line option ‘-Wunused-private-field’
make[2]: *** [src/login/CMakeFiles/topaz_connect.dir/build.make:76: src/login/CMakeFiles/topaz_connect.dir/__/common/blowfish.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:544: src/login/CMakeFiles/topaz_connect.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 11%] Linking CXX static library libspdlog.a
[ 11%] Built target spdlog
make: *** [Makefile:136: all] Error 2
```

```
c++ -v
Using built-in specs.
COLLECT_GCC=c++
COLLECT_LTO_WRAPPER=/usr/lib/gcc/aarch64-linux-gnu/11/lto-wrapper
Target: aarch64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 11.2.0-19ubuntu1' --with-bugurl=file:///usr/share/doc/gcc-11/README.Bugs --enable-languages=c,ada,c++,go,d,fortran,objc,obj-c++,m2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-11 --program-prefix=aarch64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-bootstrap --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-libquadmath --disable-libquadmath-support --enable-plugin --enable-default-pie --with-system-zlib --enable-libphobos-checking=release --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --enable-fix-cortex-a53-843419 --disable-werror --enable-checking=release --build=aarch64-linux-gnu --host=aarch64-linux-gnu --target=aarch64-linux-gnu --with-build-config=bootstrap-lto-lean --enable-link-serialization=2
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 11.2.0 (Ubuntu 11.2.0-19ubuntu1)
```

WinterSolstice8 committed:
[build] Update a bunch of "useless casts", make compiler happy

* Note, this commit doesn't fix the build entirely for GCC
dependencies still error. Even when build is successful, topaz_game crashes on login.
Please scrutinize this commit to the furthest extent possible.

WinterSolstice8 committed:
[build] Fix charentity modification

WinterSolstice8 committed:
fix typo, equiploc -> equipLoc

zach2good committed:
Merge pull request #1673 from WinterSolstice8/gcc_build_updates

[build] Update a bunch of "useless casts", make compiler happy

zach2good committed:
[chore] Various style improvements

zach2good committed:
Merge branch 'simpify_build' of github.com:LandSandBoat/server into simpify_build

zach2good committed:
[chore] Improve building of ext libs

zach2good committed:
[chore] Update and check in concurrentqueue

zach2good committed:
[chore] Fix issues with spdlog odr warnings

zach2good committed:
[chore] Fix common include relativity

zach2good committed:
[chore] Clear up latest round of warnings

zach2good committed:
[chore] cppcheck pass

zach2good committed:
[core] Fix default init of map_session_data_t

zach2good committed:
[core] Better logging for ZMQ messaging

zach2good committed:
[chore] Fix more clang warnings

zach2good committed:
[core] Revert and fix fallthrough for SmallPacket0x04E

zach2good committed:
[core] Revert and fix fallthrough for PushPacket

zach2good commented:
Ok, this should now build and run on all platforms, without errors or warnings. The build is significantly simpler. 

I've tested: Logging on, getting cutscenes, interacting with NPCs, gobbie box workflow, auction workflow, taking trusts to kuftal tunnel and fighting a bunch of crabs, casting spells, various GM commands.

Ready for review etc.

claywar reviewed (approved):
Scary one, but this is the push in the right direction 👍

WinterSolstice8 committed:
[core] fix "seen" KI bit copying

WinterSolstice8 commented:
Build works on aarch64 ubuntu 22.04 with gcc 11.2

zach2good committed:
Merge pull request #1675 from WinterSolstice8/fix_seen_KI_packet

[core] fix "seen" KI bit copying

zach2good committed:
[core] Fix MSVC release build

zach2good committed:
Merge branch 'simpify_build' of github.com:LandSandBoat/server into simpify_build

zach2good committed:
[ci] Add Win64 Release to CI builds
